### PR TITLE
Refactor Office 365 firewall setup for BusinessVM

### PR DESCRIPTION
**Pull request by vadika**
Office365 firewalling refactored to use Microsoft-provided PAC data to ease firewall passthrough and automate  application of future changes in this data.

## Checklist for things done

- [ ] Summary of the proposed changes in the PR description
- [ ] More detailed description in the commit message(s)
- [ ] Commits are squashed into relevant entities - avoid a lot of minimal dev time commits in the PR
- [ ] [Contribution guidelines](https://github.com/tiiuae/ghaf/blob/main/CONTRIBUTING.md) followed
- [ ] Ghaf documentation updated with the commit - https://tiiuae.github.io/ghaf/
- [ ] PR linked to architecture documentation and requirement(s) (ticket id)
- [ ] Test procedure described (or includes tests). Select one or more:
  - [ ] Tested on Lenovo X1 `x86_64`
  - [ ] Tested on Jetson Orin NX or AGX `aarch64`
  - [ ] Tested on Polarfire `riscv64`
- [ ] Author has run `make-checks` and it passes
- [ ] All automatic Github Action checks pass - see [actions](https://github.com/tiiuae/ghaf/actions)
- [ ] Author has added reviewers and removed PR draft status
- [ ] Change requires full re-installation
- [ ] Change can be updated with `nixos-rebuild ... switch`

## Instructions for Testing

- [ ] List all targets that this applies to:
- [ ] Is this a new feature
  - [ ] List the test steps to verify:
- [ ] If it is an improvement how does it impact existing functionality?

### CODE REVIEW

**Comment by milva-unikie** [comment]
I can open youtube.com (and many other pages) in Trusted browser. Firewall seems to be not working at all.

**Comment by milva-unikie** [comment]
I can still open Youtube in Trusted browser.

**Comment by milva-unikie** [comment]
And now everything is blocked. Even Teams and Outlook do not work.

File: modules/reference/appvms/business.nix [hunk 1]
```diff
@@ -11,6 +11,62 @@ let
...
+    }
+
+    generate_iptables_rules() {
+        echo "# Office 365 Firewall Rules"
+        echo
+
+        jq -r '.[] | select(.category == "Optimize" or .category == "Allow" or .category == "Default") | .ips[]?' "$ENDPOINTS_FILE" | sort -u | while read -r ip; do
+            if [[ $ip == *":"* ]]; then
+                echo "ip6tables -I OUTPUT -d $ip -j ACCEPT"
+                echo "ip6tables -I INPUT -d $ip -j ACCEPT"
```
Comment (enesoztrk) [inline modules/reference/appvms/business.nix:41]: we need to check the source IP for input. `ip6tables -I INPUT -s $ip -j ACCEPT`

File: modules/reference/appvms/business.nix [hunk 2]
```diff
@@ -11,6 +11,62 @@ let
...
+            else
+                echo "iptables -I OUTPUT -d $ip -j ACCEPT"
+                echo "iptables -I INPUT -d $ip -j ACCEPT"
```
Comment (enesoztrk) [inline modules/reference/appvms/business.nix:44]: we need to check the source IP for input. `iptables -I INPUT -s $ip -j ACCEPT`

File: modules/reference/appvms/business.nix [hunk 3]
```diff
@@ -11,6 +11,62 @@ let
...
+        done
+
+        echo
+
+        jq -r '.[] | select(.category == "Optimize" or .category == "Allow" or .category == "Default") | .urls[]?' "$ENDPOINTS_FILE" | sort -u | while read -r url; do
+            processed_url=$(preprocess_url "$url")
+            if [[ "$processed_url" != "" ]]; then
+                echo "iptables -I OUTPUT -p tcp --dport 80 -m string --string \"$processed_url\" --algo bm -j ACCEPT"
+                echo "iptables -I INPUT -p tcp --dport 80 -m string --string \"$processed_url\" --algo bm -j ACCEPT"
+                echo "iptables -I OUTPUT -p tcp --dport 443 -m string --string \"$processed_url\" --algo bm -j ACCEPT"
```
Comment (enesoztrk) [inline modules/reference/appvms/business.nix:55]: Most of the websites use HTTPS, which means that the payloads are encrypted. This method does not work.

File: modules/reference/appvms/business.nix [hunk 4]
```diff
@@ -11,6 +11,62 @@ let
...
+                echo "iptables -I INPUT -p tcp --dport 443 -m string --string \"$processed_url\" --algo bm -j ACCEPT"
```
Comment (enesoztrk) [inline modules/reference/appvms/business.nix:56]: Same as previous comment. In addition, `--sport` should have been used rather than `--dport`.

**Review by enesoztrk — CHANGES_REQUESTED** [review]

**Comment by vadika** [comment]
Dropping this PR in favor ot https://github.com/tiiuae/ghaf/pull/811

**Pull request closed without merging**